Sources/Widgets/Widgets3D/ShapeWidget/example/index.js

```diff
@@ -276,10 +278,45 @@ reader
       }
     );
 
+    const updateWidgetVisibility = (widget, slicePos, i, widgetIndex) => {
+      /* testing if the widget is on the slice and has been placed to modify visibility */
+      const widgetVisibility =
+        widget.getWidgetState().getPoint1Handle().getOrigin()[i] ===
+          slicePos[i] ||
+        !scene.widgetManager.getWidgets()[widgetIndex].getPoint1();
+      widget.setVisibility(widgetVisibility);
+    };
+
+    const updateWidgetsVisibility = (position, slicingMode) => {
+      updateWidgetVisibility(widgets.rectangleWidget, position, slicingMode, 0);
+      updateWidgetVisibility(widgets.ellipseWidget, position, slicingMode, 1);
+      updateWidgetVisibility(widgets.circleWidget, position, slicingMode, 2);
+    };
+
     const update = () => {
       const slicingMode = image.imageMapper.getSlicingMode() % 3;
 
       if (slicingMode > -1) {
+        const ijk = [0, 0, 0];
+        const position = [0, 0, 0];
+        const normal = [0, 0, 0];
+
+        // position
+        ijk[slicingMode] = image.imageMapper.getSlice();
+        data.indexToWorldVec3(ijk, position);
+
+        // circle/slice normal
+        ijk[slicingMode] = 1;
+        data.indexToWorldVec3(ijk, normal);
+        vtkMath.subtract(normal, data.getOrigin(), normal);
+        vtkMath.normalize(normal);
```

Comment:
What's l300 to l312 used for? Was that to compute the normal but you're not setting it anymore?

If yes, please revert, and note that you could access it in the future using `imageMapper.getSlicingModeNormal()`